src/app/ApplicationTemplate.Shared.Views/Controls/AppBarBackButton.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Reactive.Linq;
+using System.Windows.Input;
+using Chinook.DynamicMvvm;
+using Chinook.SectionsNavigation;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.UI.Dispatching;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+
+namespace ApplicationTemplate.Views.Controls;
+
+public sealed partial class AppBarBackButton : AppBarButton
+{
+	private IDisposable _backButtonVisibilitySubscription;
+
+	public AppBarBackButton()
+	{
+		Icon = new FontIcon()
+		{
+			// Value in XAML (&#xE72B).
+			Glyph = "\xE72B",
+		};
+		Command = BuildClickCommand();
+
+		Loaded += OnLoaded;
+		Unloaded += OnUnloaded;
+	}
+
+	private static IServiceProvider ServiceProvider => ViewModelBase.DefaultServiceProvider;
+
+	private static ISectionsNavigator SectionsNavigator => ServiceProvider.GetRequiredService<ISectionsNavigator>();
+
+	private static IDynamicCommandBuilderFactory CommandBuilderFactory { get; } = new DynamicCommandBuilderFactory(c => c
+		.CatchErrors(ServiceProvider.GetRequiredService<IDynamicCommandErrorHandler>())
+		.WithLogs(ServiceProvider.GetRequiredService<ILogger<IDynamicCommand>>())
+		.DisableWhileExecuting()
+	);
+
+	private static ICommand BuildClickCommand() => CommandBuilderFactory
+		.CreateFromTask("Click", async (ct) => await SectionsNavigator.NavigateBackOrCloseModal(ct))
```

Comment:
Wouldn't it be better to request a back on a `IBackButtonSource`?

Reply:
Added a comment and will see in the future as said in Teams.